incubator/mariadb-cluster/templates/slave-statefulset.yaml

```diff
@@ -134,24 +149,22 @@ spec:
 {{ toYaml .Values.metrics.resources | indent 10 }}
 {{- end }}
       volumes:
-      - name: data
-        emptyDir: {}
----
-apiVersion: autoscaling/v1
-kind: HorizontalPodAutoscaler
-metadata:
-  name: {{ template "slave.fullname" . }}
-  labels:
-    app: "{{ template "name" . }}"
-    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
-    component: "slave"
-    release: "{{ .Release.Name }}"
-    heritage: "{{ .Release.Service }}"
-spec:
-  minReplicas: {{ .Values.slave.hpa.min }}
-  maxReplicas: {{ .Values.slave.hpa.max }}
-  scaleTargetRef:
-    apiVersion: extensions/v1beta1
-    kind: Deployment
-    name: {{ template "slave.fullname" . }}
-  targetCPUUtilizationPercentage: {{ .Values.slave.hpa.target.cpuPercentage }}
+        - name: config
```

Comment:
add the following around the `config` volume block

```
{{- if .Values.slave.config }}
{{- end }}
```

And restore the `emtyDir` volume for the `data` when persistence is disabled (same as the master)